ironfish-cli/src/commands/service/snapshot.ts

```diff
@@ -0,0 +1,201 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Assert, AsyncUtils, FileUtils } from '@ironfish/sdk'
+import { CliUx, Flags } from '@oclif/core'
+import { S3 } from 'aws-sdk'
+import crypto from 'crypto'
+import fsAsync from 'fs/promises'
+import os from 'os'
+import path from 'path'
+import tar from 'tar'
+import { IronfishCommand } from '../../command'
+import { RemoteFlags } from '../../flags'
+import { ProgressBar } from '../../types'
+
+export default class CreateSnapshot extends IronfishCommand {
+  static hidden = true
+
+  static description = `Upload chain snapshot to a public bucket`
+
+  static flags = {
+    ...RemoteFlags,
+    bucket: Flags.string({
+      char: 'b',
+      parse: (input: string) => Promise.resolve(input.trim()),
+      required: false,
+      description: 'S3 bucket to upload snapshot to',
+    }),
+    path: Flags.string({
+      char: 'p',
+      parse: (input: string): Promise<string> => Promise.resolve(input.trim()),
+      required: false,
+      description: 'The path where the snapshot should be saved',
+    }),
+    maxBlocksPerChunk: Flags.integer({
+      char: 'm',
+      required: false,
+      default: isNaN(Number(process.env.MAX_BLOCKS_PER_SNAPSHOT_CHUNK))
+        ? 1000
+        : Number(process.env.MAX_BLOCKS_PER_SNAPSHOT_CHUNK),
+      description: 'The max number of blocks per file in the zipped snapshot',
+    }),
+  }
+
+  async start(): Promise<void> {
+    const { flags } = await this.parse(CreateSnapshot)
+
+    const bucket = (flags.bucket || process.env.IRONFISH_SNAPSHOT_BUCKET || '').trim()
+
+    let exportDir
+
+    if (flags.path) {
+      exportDir = this.sdk.fileSystem.resolve(flags.path)
+    } else {
+      try {
+        exportDir = await fsAsync.mkdtemp(`${os.tmpdir()}${path.sep}`)
```

Comment:
It might be worth investigating whether it's possible to write large files into the os's tmp directory -- considering the snapshots can be 10s of gigabytes, they might cap out the temp directory on linux/windows. I'm not too familiar with limitations of temp directories though

Reply:
After researching this for a few minutes, there doesn't seem to be any file size limitation for temp directories for any major OS. Additionally, the temp directory is limited by either the overall hard disk space (Windows) or by the partition where it resides (Unix-based/Linux).

Reply:
Okay, sounds good, let's try this out then and we can follow up if we get any inbound issues from users. This should mostly be used by service operators like us, anyway